### DISCUSSION TITLE
PythonScriptLinesGenerator.py or PythonScriptLinesGenerator2.py

### DISCUSSION THREAD

**Discussion by tom2rd** [LubomirJagos/FreeCAD-OpenEMS-Export#39] [General]
Thank you for good macros for OpenEMS and FreeCAD.

I'm trying to running with python, and find the error in  "FreeCAD-OpenEMS-Export-main\utilsOpenEMS\ScriptLinesGenerator\PythonScriptLinesGenerator.py"
and 2.py

def mesh():
  x, y, z

maybe change as follows,

class Mesh:
    def __init__(self):
        self.x = np.array([])
        self.y = np.array([])
        self.z = np.array([])

mesh = Mesh()

--------------- 

def getCoordinateSystemScriptLines(self):
        genScript = ""

        genScript += "#######################################################################################################################################\n"
        genScript += "# COORDINATE SYSTEM\n"
        genScript += "#######################################################################################################################################\n"

        """ # Till now not used, just using rectangular coordination type, cylindrical MUST BE IMPLEMENTED!
        gridCoordsType = self.getModelCoordsType()
        if (gridCoordsType == "rectangular"):
            genScript += "CSX = InitCSX('CoordSystem',0); # Cartesian coordinate system.\n"
        elif (gridCoordsType == "cylindrical"):
            genScript += "CSX = InitCSX('CoordSystem',1); # Cylindrical coordinate system.\n"
        else:
            genScript += "%%%%%% ERROR GRID COORDINATION SYSTEM TYPE UNKNOWN"				
        """
        genScript += "FDTD.SetCoordSystem(0) # Cartesian coordinate system.\n"
        genScript +="class Mesh:\n"
        genScript +="\tdef __init__(self):\n"
        genScript +="\t\tself.x = np.array([])\n"
        genScript +="\t\tself.y = np.array([])\n"
        genScript +="\t\tself.z = np.array([])\n"
        genScript +="mesh = Mesh()\n"
        #genScript += "def mesh():\n"
        #genScript += "\tx,y,z\n"
        genScript += "\n"
        genScript += "mesh.x = np.array([]); # mesh variable initialization (Note: x y z implies type Cartesian).\n"
        genScript += "mesh.y = np.array([]);\n"
        genScript += "mesh.z = np.array([]);\n"
        genScript += "openEMS_grid = CSX.GetGrid()\n"
        genScript += "openEMS_grid.SetDeltaUnit(unit) # First call with empty mesh to set deltaUnit attribute.\n"
        genScript += "\n"

        return genScript

**Comment by LubomirJagos42** [comment]
Hi, thanks for notification, I will check that, I made cylindrical coords working few months ago, I was using octave interface and I thought that I made it running also for python interface, but seems that something left there wrong.
Will look at it and repair that, initially I think that uncommenting that stuff should be enough but I will see.

**Reply by tom2rd** [reply]
Thank for reply.
I'm really interested in this project, So, If you already find issues, please teach me. I'll try them.

**Reply by LubomirJagos42** [reply]
yeah, this GUI over openEMS is really functional, it's like you can write python scripts and so you will make it much better, but if you sit in front of openEMS every time after 3-4 months then generator from GUI is much more better due you don't remember what you wanted to do :D :D
just joking, but this GUI is like to provide some user experience on advantage level to somehow make openEMS rival of HFSS
